lib/vrl/stdlib/src/parse_syslog.rs

```diff
@@ -57,7 +58,14 @@ impl Expression for ParseSyslogFn {
         let value = self.value.resolve(ctx)?;
         let message = value.try_bytes_utf8_lossy()?;
 
-        let parsed = syslog_loose::parse_message_with_year_exact(&message, resolve_year)?;
+        let parsed = match ctx.timezone() {
+            TimeZone::Local => {
+                syslog_loose::parse_message_with_year_exact_tz(&message, resolve_year, Some(Local))
+            }
+            TimeZone::Named(tz) => {
+                syslog_loose::parse_message_with_year_exact_tz(&message, resolve_year, Some(*tz))
+            }
+        }?;
```

Comment:
This doesn't work 🤔 
```
error[E0308]: `match` arms have incompatible types
  --> lib/vrl/stdlib/src/parse_syslog.rs:63:36
   |
61 |           let tz = match ctx.timezone() {
   |  __________________-
62 | |             TimeZone::Local => Local,
   | |                                ----- this is found to be of type `chrono::Local`
63 | |             TimeZone::Named(tz) => *tz,
   | |                                    ^^^ expected struct `chrono::Local`, found enum `chrono_tz::timezones::Tz`
64 | |         };
   | |_________- `match` arms have incompatible types

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `vrl-stdlib`
```

Reply:
Ah! Right. That's because [`parse_message_with_year_exact_tz`](https://docs.rs/syslog_loose/0.14.0/syslog_loose/fn.parse_message_with_year_exact_tz.html) expects a type that adheres to the `TimeZone` trait. Both of these types do, but right now, each match arm returns a different concrete type, which is not allowed.

One solution is to store a `Box<TimeZone>` into `tz`. However, from reading the `syslog_loose` source, it appears that `Some(chrono::Local)` and `None` are equivalent when calling `parse_message_with_year_exact_tz`, so a better alternative in this case would be:

```suggestion
        let tz = match ctx.timezone() {
            TimeZone::Local => None,
            TimeZone::Named(tz) => Some(*tz),
        };
        
        syslog_loose::parse_message_with_year_exact_tz(&message, resolve_year, tz)
```